superchain/superchain_test.go

```diff
@@ -395,3 +395,90 @@ func testNetworkUpgradeTimestampOffset(l2GenesisTime uint64, blockTime uint64, u
 		}
 	}
 }
+
+func TestSuperchainConfigUnmarshaling(t *testing.T) {
+	rawYAML := `
+name: Mickey Mouse
+l1:
+  chain_id: 314
+  public_rpc: https://disney.com
+  explorer: https://disneyscan.io
+
+protocol_versions_addr: "0x252CbE9517F731C618961D890D534183822dcC8d"
+superchain_config_addr: "0x02d91Cf852423640d93920BE0CAdceC0E7A00FA7"
+
+canyon_time: 1
+delta_time: 2
+ecotone_time: 3
+fjord_time: 
+`
+
+	s := SuperchainConfig{}
+	err := unMarshalSuperchainConfig([]byte(rawYAML), &s)
+	require.NoError(t, err)
+
+	require.Equal(t, "Mickey Mouse", s.Name)
+	require.Equal(t, SuperchainL1Info{
+		ChainID:   314,
+		PublicRPC: "https://disney.com",
+		Explorer:  "https://disneyscan.io",
+	}, s.L1)
+	require.Equal(t, "0x252cbe9517f731c618961d890d534183822dcc8d", s.ProtocolVersionsAddr.String())
+	require.Equal(t, "0x02d91cf852423640d93920be0cadcec0e7a00fa7", s.SuperchainConfigAddr.String())
+	require.Equal(t, uint64Ptr(uint64(1)), s.hardForkDefaults.CanyonTime)
+	require.Equal(t, uint64Ptr(uint64(2)), s.hardForkDefaults.DeltaTime)
+	require.Equal(t, uint64Ptr(uint64(3)), s.hardForkDefaults.EcotoneTime)
+	require.Nil(t, s.hardForkDefaults.FjordTime)
+}
+
+func TestHardForkOverridesAndDefaults(t *testing.T) {
+
+	defaultCanyonTime := uint64(3)
+
+	// Set a ChainConfig to unmarshal into
+	// which already has a default value set
+	s := SuperchainConfig{
+		hardForkDefaults: HardForkConfiguration{
+			CanyonTime: &defaultCanyonTime,
+		}}
+
+	t.Run("override: unmarshal with an override", func(t *testing.T) {
```

Comment:
Would be great to run these 3 tests in a test table instead. We could then also add 3 more cases for when the superchain `CanyonTime` field is nil. And one more complete test would also be good, where all currently known fork fields are set either in the superchain config, or in the chain config.

Reply:
[97db949](https://github.com/ethereum-optimism/superchain-registry/pull/101/commits/97db9495695936c9ba084aff1720cd8a1d5279d5)

Reply:
[2dcb036](https://github.com/ethereum-optimism/superchain-registry/pull/101/commits/2dcb0367ff3a685e44eed8de424de0234d6d6a6a)